packages/backend/src/peripherals/database/EventRepository.ts

```diff
@@ -59,30 +68,23 @@ export class EventRepository extends BaseRepository {
     from: UnixTime = new UnixTime(0),
   ): Promise<EventRecordAggregated[]> {
     const knex = await this.knex()
-    const rows = (await knex('events')
-      .where('project_id', projectId.toString())
-      .where('unix_timestamp', '>=', from.toDate())
-      .select(
-        knex.raw(
-          `date_trunc('${granularity}', unix_timestamp) AS unix_timestamp`,
-        ),
-        'event_name',
-        'project_id',
-        knex.raw('COUNT(*) as count'),
-      )
-      .groupBy(
-        'event_name',
-        'project_id',
-        knex.raw(`date_trunc('${granularity}', unix_timestamp)`),
-      )
-      //todo fix typing
-      .orderBy(['unix_timestamp', 'event_name'])) as unknown as {
-      unix_timestamp: Date
-      event_name: string
-      project_id: string
-      count: number
-    }[]
-    return rows.map(toRecordAggregated)
+
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+    if (granularity !== 'hour' && granularity !== 'day') {
+      throw new Error('Programmer error or SQL injection')
+    }
+
+    const query =
+      granularity === 'hour'
```

Comment:
why not just pass granuality directly?

Reply:
@krzkaczor there was an error with encoding, I fixed it by changing the column name, you can take a look now